packages/twenty-front/src/pages/settings/data-model/SettingsObjectEdit.tsx

```diff
@@ -81,10 +82,16 @@ export const SettingsObjectEdit = () => {
     formValues: SettingsDataModelObjectEditFormValues,
   ) => {
     let values = formValues;
-    if (
-      formValues.isLabelSyncedWithName === true ||
-      activeObjectMetadataItem.isLabelSyncedWithName === true
-    ) {
+    const dirtyFieldKeys = Object.keys(
+      formConfig.formState.dirtyFields,
+    ) as (keyof SettingsDataModelObjectEditFormValues)[];
+    const shouldComputeNamesFromLabels: boolean = dirtyFieldKeys.includes(
+      IS_LABEL_SYNCED_WITH_NAME_LABEL,
+    )
+      ? (formValues.isLabelSyncedWithName as boolean)
```

Comment:
```suggestion
      ? !!formValues.isLabelSyncedWithName
```
or
```suggestion
      ? (formValues.isLabelSyncedWithName ?? false)
```
would be better than type assertion imho